sdk/search/azure-search/src/main/java/com/azure/search/documents/SearchApiKeyPipelinePolicy.java

```diff
@@ -12,31 +13,37 @@
 import java.util.Objects;
 
 /**
- * Policy that adds the Cognitive Search Service api-key into the request's Authorization header.
+ * Pipeline policy that uses an {@link AzureKeyCredential} to set the {@code api-key} header which authorizes requests
+ * sent to the Azure Search service.
+ *
+ * <p>
+ * Requests sent with this pipeline policy are required to use {@code HTTPS}. If the request isn't using {@code HTTPS}
+ * an exception will be thrown to prevent leaking the key.
  */
 public final class SearchApiKeyPipelinePolicy implements HttpPipelinePolicy {
```

Comment:
So this means that every client needs to implement it's own policy, is that correct?
Wondering if we could add a generic policy to Azure Core too

Reply:
I was discussing this with @JonathanGiles and we didn't strongly lean on one design or the other. I left the library specific policies in place given they are in preview and adding this into Azure Core would likely require it being part of the public API which could GA it sooner (Java doesn't have a good friends assembly or internal that .NET leverages).

@JonathanGiles any concern around making this a generic Azure Core policy that uses the credential and a header string? We do something conceptually similar with the `RetryPolicy` where we can pass a retry header to look for on the response.

Reply:
>I left the library specific policies in place given they are in preview

Note that Text Analytics is going to GA in April, so if we could add the policy to Core now, that will be great for TA.

Reply:
Couldn't the library specific policies be internal to the package, or added under implementation? Then you can always update them to use an implementation exposed by core at some point in the future.

Reply:
Yes, this could be done but goes a bit against the response above around why the policy is in the public API.